src/util/__tests__/ChromeHelper.test.ts

```diff
@@ -0,0 +1,154 @@
+jest.mock('../chrome', () => ({
+  launch: jest.fn(),
+}));
+
+jest.mock('../../Chrome');
+
+import { ChromeHelper } from '../ChromeHelper';
+import * as chromeUtil from '../chrome';
+
+const getMockedBrowser = () => ({
+  browser: {
+    kill: jest.fn(),
+    port: 8675,
+  },
+  cdp: {
+    close: jest.fn(),
+    Target: {
+      closeTarget: jest.fn(),
+    },
+  },
+});
+
+const getMockedTab = (targetId) => () => ({
+  tab: {},
+  targetId,
+});
+
+describe('ChromeHelper', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  })
+
+  describe('#start', () => {
+    it('should not start the browser again if it has already been started', () => {
+      const mockedBrowser = getMockedBrowser();
+      chromeUtil.defaultFlags = {};
+      chromeUtil.launch = jest.fn(() => Promise.resolve(mockedBrowser));
+      chromeUtil.createTab = jest.fn(getMockedTab('targetId'));
+
+      const helper = new ChromeHelper({});
+      return helper.start().then(() => {
+        return helper.start();
+      }).then(() => {
+        helper.onTabClose('tabTargetId');
+        expect(chromeUtil.launch).toHaveBeenCalledWith(chromeUtil.defaultFlags, true);
+        expect(chromeUtil.launch).toHaveBeenCalledTimes(1);
+      });
+    })
+  });
+
+  describe('#onTabClose', () => {
+    // TODO: probably want a better error message for this case?
```

Comment:
It's not likely that someone would call `onTabClose` directly but it might be good to warn / error more gracefully if `browserStartingPromise` isn't set yet.